bugsnag/wsgi/middleware.py

```diff
@@ -21,11 +21,12 @@ def add_wsgi_request_data_to_notification(notification):
 
     environ = notification.request_config.wsgi_environ
     request = Request(environ)
+    path = request_path(environ)
 
-    notification.context = "%s %s" % (request.method, request_path(environ))
+    notification.context = "%s %s" % (request.method, path)
     notification.set_user(id=request.remote_addr)
     notification.add_tab("request", {
-        "url": request.path_url,
+        "url": request.application_url + path,
```

Comment:
Do we know that `application_url` is safe to use?

Reply:
I'll do some reading to make sure but I assume it will be safe because if it's badly encoded I don't think you would be able reach the webserver in the first place, though I'm not sure if this is possible to pass from the client

Reply:
I'm not sure about this to be honest — the `application_url` will be built using the `Host` header (if present), which is what's recommended in [the WSGI PEP](https://www.python.org/dev/peps/pep-3333/#url-reconstruction) but means that it's sent from the client so could have the same issues

Usually though the `Host` header would be set by the webserver talking to Python so the client version would be ignored, but that's not a guarantee so this could have the same problem

<details>
<summary>Github, for example, will try to redirect you if you send a bad `Host`</summary>

```
$ curl 'https://github.com' -H 'Host: google.com' -v
...
< HTTP/1.1 301 Moved Permanently
< Content-length: 0
< Location: https://github.com/
...
```

</details>

<details>
<summary>Python.org rejects the request completely</summary>

```
$ curl https://www.python.org -H 'Host: google.com'
<html>
<head>
<title>Fastly error: unknown domain google.com</title>
</head>
<body>
<p>Fastly error: unknown domain: google.com. Please check that this domain has been added to a service.</p>
</body>
</html>
```
</details>

Does it make sense to implement the [URL reconstruction algorithm](https://www.python.org/dev/peps/pep-3333/#url-reconstruction) ourselves, ignoring `HTTP_HOST`, if accessing `request.application_url` fails?

I don't think we can change the default of `request.application_url` because without the `Host` header, requests that are fronted by something like NGINX will be to `localhost` or an internal IP rather than the actual URL

Reply:
I think on balance what you have is appropriate as it's much less likely that the host headers will be badly encoded and different from the actual URL used to reach the server in the first place.